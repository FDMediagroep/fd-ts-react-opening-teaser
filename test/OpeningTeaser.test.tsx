import TestRenderer from 'react-test-renderer';
import OpeningTeaser from '../src/OpeningTeaser';
import React from 'react';
import uniqid from 'uniqid';

describe('OpeningTeaser', () => {
    const teaserData: any = {
        id: "1286721",
        baseUrl: "https://fd.nl",
        url: "https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe",
        image: {
            src: 'https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663',
            alt: 'Mario Draghi, president of the European Central Bank (ECB), departs a rates decision news conference in Frankfurt, Germany, on Thursday, Dec. 13, 2018. Draghi??said risks to the euro-area economy are worsening even as he called time on the??European Central Bank???s??flagship deflation-fighting tool. Photographer: Alex Kraus/Bloomberg'
        },
        sourceSets: [{
            media: "(max-width:640px)",
            srcSet: `https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663`
        }, {
            media: "(max-width:860px)",
            srcSet: `https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=723&h=406&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663`
        }, {
            media: "(min-width:861px)",
            srcSet: `https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=719&h=404&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663`
        }],
        figCaption: "Amsterdam doet ertoe",
        subject: "Buitenland",
        comments: 1,
        readableAge: "34 min",
        date: "17 Nov 1981",
        title: "Amsterdam doet ertoe",
        description: "Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.",
        related: [{
            longread: true,
            title: '"Als vrouw moet je altijd onafhankelijk zijn"',
            url: 'https://dev.fd.nl/profiel/1235280/als-vrouw-moet-je-onafhankelijk-zijn',
            uuid: uniqid()
        }, {
            title: 'President van de Amerikaanse Fed hamert op onafhankelijkheid',
            url: 'https://dev.fd.nl/economie-politiek/1235274/president-van-de-amerikaanse-fed-hamert-op-onafhankelijkheid',
            uuid: uniqid()
        }]
    };

    test('renders correctly', () => {
        let openingTeaser = TestRenderer.create(<OpeningTeaser {...teaserData}/>);
        expect(openingTeaser.toJSON()).toMatchSnapshot();
    });

    test('renders bookmarked correctly', () => {
        let openingTeaser = TestRenderer.create(<OpeningTeaser {...teaserData} bookmarked={true}/>);
        expect(openingTeaser.toJSON()).toMatchSnapshot();
    });

    test('renders multiple comments correctly', () => {
        let openingTeaser = TestRenderer.create(<OpeningTeaser {...teaserData} comments={99}/>);
        expect(openingTeaser.toJSON()).toMatchSnapshot();
    });

    test('renders longread correctly', () => {
        let openingTeaser = TestRenderer.create(<OpeningTeaser {...teaserData} cardStyle='longread'/>);
        expect(openingTeaser.toJSON()).toMatchSnapshot();
    });

    test('renders persoonlijk correctly', () => {
        let openingTeaser = TestRenderer.create(<OpeningTeaser {...teaserData} cardStyle='persoonlijk'/>);
        expect(openingTeaser.toJSON()).toMatchSnapshot();
    });

    test('renders persoonlijk longread correctly', () => {
        let openingTeaser = TestRenderer.create(<OpeningTeaser {...teaserData} cardStyle='persoonlijk longread'/>);
        expect(openingTeaser.toJSON()).toMatchSnapshot();
    });

});
