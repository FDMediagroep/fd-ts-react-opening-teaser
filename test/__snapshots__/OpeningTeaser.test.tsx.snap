// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningTeaser renders bookmarked correctly 1`] = `
<div
  cardStyle="default"
  className="fd-card fd-opening-teaser default"
  id="1286721"
>
  <a
    href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
  >
    <figure
      className="fd-teaser-figure"
    >
      <picture>
        <source
          media="(max-width:640px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(max-width:860px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=723&h=406&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(min-width:861px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=719&h=404&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <img
          alt="Mario Draghi, president of the European Central Bank (ECB), departs a rates decision news conference in Frankfurt, Germany, on Thursday, Dec. 13, 2018. Draghi??said risks to the euro-area economy are worsening even as he called time on the??European Central Bank???s??flagship deflation-fighting tool. Photographer: Alex Kraus/Bloomberg"
          src="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
      </picture>
    </figure>
    <div
      className="meta"
    >
      <span
        className="prefix"
      >
        Buitenland
      </span>
      <time>
        34 min
      </time>
    </div>
    <div
      className="text-container"
    >
      <h1
        className="fd-typography opening-teaser-h"
        textStyle="opening-teaser-h"
      >
        Amsterdam doet ertoe
      </h1>
      <p
        className="intro"
      >
        Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.
      </p>
    </div>
  </a>
  <nav
    className="related"
  >
    <a
      className="related-link longread"
      href="https://dev.fd.nl/profiel/1235280/als-vrouw-moet-je-onafhankelijk-zijn"
    >
      <span
        className="related-title"
      >
        "Als vrouw moet je altijd onafhankelijk zijn"
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
    <a
      className="related-link"
      href="https://dev.fd.nl/economie-politiek/1235274/president-van-de-amerikaanse-fed-hamert-op-onafhankelijkheid"
    >
      <span
        className="related-title"
      >
        President van de Amerikaanse Fed hamert op onafhankelijkheid
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
  </nav>
  <footer
    className="fd-teaser-footer"
  >
    <menu>
      <span
        className="placeholder"
      />
      <span
        className="share-social"
      >
        <i
          className="icon-share-alt"
          data-id="1286721"
          data-section="Buitenland"
          data-title="Amsterdam doet ertoe"
          data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
          title="Delen"
        />
        <span
          className="icon-label"
        >
          Delen
        </span>
      </span>
      <span
        className="favorite"
        data-addurl="/add-favorite"
        data-articleid="1286721"
        data-deleteurl="/delete-favorite"
        data-key="1286721"
        data-title="Amsterdam doet ertoe"
        data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
        id="favorite1286721"
        title="Bewaren"
      >
        <i
          className="icon-bookmark"
          title="Bewaren"
        />
        <i
          className="icon-bookmark1"
          title="Verwijder van bewaarlijst"
        />
        <span
          className="icon-label"
        >
          Bewaren
        </span>
      </span>
      <a
        className="comments-link"
        data-articleid="1286721"
        href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe#comments"
      >
        <i
          className="icon-comment-alt"
        />
        <span
          className="comments-count"
        >
          1
        </span>
        <span
          className="comments-suffix"
        >
          reactie
        </span>
      </a>
    </menu>
  </footer>
</div>
`;

exports[`OpeningTeaser renders correctly 1`] = `
<div
  cardStyle="default"
  className="fd-card fd-opening-teaser default"
  id="1286721"
>
  <a
    href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
  >
    <figure
      className="fd-teaser-figure"
    >
      <picture>
        <source
          media="(max-width:640px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(max-width:860px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=723&h=406&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(min-width:861px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=719&h=404&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <img
          alt="Mario Draghi, president of the European Central Bank (ECB), departs a rates decision news conference in Frankfurt, Germany, on Thursday, Dec. 13, 2018. Draghi??said risks to the euro-area economy are worsening even as he called time on the??European Central Bank???s??flagship deflation-fighting tool. Photographer: Alex Kraus/Bloomberg"
          src="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
      </picture>
    </figure>
    <div
      className="meta"
    >
      <span
        className="prefix"
      >
        Buitenland
      </span>
      <time>
        34 min
      </time>
    </div>
    <div
      className="text-container"
    >
      <h1
        className="fd-typography opening-teaser-h"
        textStyle="opening-teaser-h"
      >
        Amsterdam doet ertoe
      </h1>
      <p
        className="intro"
      >
        Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.
      </p>
    </div>
  </a>
  <nav
    className="related"
  >
    <a
      className="related-link longread"
      href="https://dev.fd.nl/profiel/1235280/als-vrouw-moet-je-onafhankelijk-zijn"
    >
      <span
        className="related-title"
      >
        "Als vrouw moet je altijd onafhankelijk zijn"
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
    <a
      className="related-link"
      href="https://dev.fd.nl/economie-politiek/1235274/president-van-de-amerikaanse-fed-hamert-op-onafhankelijkheid"
    >
      <span
        className="related-title"
      >
        President van de Amerikaanse Fed hamert op onafhankelijkheid
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
  </nav>
  <footer
    className="fd-teaser-footer"
  >
    <menu>
      <span
        className="placeholder"
      />
      <span
        className="share-social"
      >
        <i
          className="icon-share-alt"
          data-id="1286721"
          data-section="Buitenland"
          data-title="Amsterdam doet ertoe"
          data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
          title="Delen"
        />
        <span
          className="icon-label"
        >
          Delen
        </span>
      </span>
      <span
        className="favorite"
        data-addurl="/add-favorite"
        data-articleid="1286721"
        data-deleteurl="/delete-favorite"
        data-key="1286721"
        data-title="Amsterdam doet ertoe"
        data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
        id="favorite1286721"
        title="Bewaren"
      >
        <i
          className="icon-bookmark"
          title="Bewaren"
        />
        <i
          className="icon-bookmark1"
          title="Verwijder van bewaarlijst"
        />
        <span
          className="icon-label"
        >
          Bewaren
        </span>
      </span>
      <a
        className="comments-link"
        data-articleid="1286721"
        href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe#comments"
      >
        <i
          className="icon-comment-alt"
        />
        <span
          className="comments-count"
        >
          1
        </span>
        <span
          className="comments-suffix"
        >
          reactie
        </span>
      </a>
    </menu>
  </footer>
</div>
`;

exports[`OpeningTeaser renders longread correctly 1`] = `
<div
  cardStyle="longread"
  className="fd-card fd-opening-teaser longread"
  id="1286721"
>
  <a
    href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
  >
    <figure
      className="fd-teaser-figure"
    >
      <picture>
        <source
          media="(max-width:640px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(max-width:860px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=723&h=406&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(min-width:861px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=719&h=404&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <img
          alt="Mario Draghi, president of the European Central Bank (ECB), departs a rates decision news conference in Frankfurt, Germany, on Thursday, Dec. 13, 2018. Draghi??said risks to the euro-area economy are worsening even as he called time on the??European Central Bank???s??flagship deflation-fighting tool. Photographer: Alex Kraus/Bloomberg"
          src="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
      </picture>
    </figure>
    <div
      className="meta"
    >
      <span
        className="prefix"
      >
        Buitenland
      </span>
      <time>
        34 min
      </time>
    </div>
    <div
      className="text-container"
    >
      <h1
        className="fd-typography opening-teaser-h"
        textStyle="opening-teaser-h"
      >
        Amsterdam doet ertoe
      </h1>
      <p
        className="intro"
      >
        Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.
      </p>
    </div>
  </a>
  <nav
    className="related"
  >
    <a
      className="related-link longread"
      href="https://dev.fd.nl/profiel/1235280/als-vrouw-moet-je-onafhankelijk-zijn"
    >
      <span
        className="related-title"
      >
        "Als vrouw moet je altijd onafhankelijk zijn"
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
    <a
      className="related-link"
      href="https://dev.fd.nl/economie-politiek/1235274/president-van-de-amerikaanse-fed-hamert-op-onafhankelijkheid"
    >
      <span
        className="related-title"
      >
        President van de Amerikaanse Fed hamert op onafhankelijkheid
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
  </nav>
  <footer
    className="fd-teaser-footer"
  >
    <menu>
      <span
        className="placeholder"
      />
      <span
        className="share-social"
      >
        <i
          className="icon-share-alt"
          data-id="1286721"
          data-section="Buitenland"
          data-title="Amsterdam doet ertoe"
          data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
          title="Delen"
        />
        <span
          className="icon-label"
        >
          Delen
        </span>
      </span>
      <span
        className="favorite"
        data-addurl="/add-favorite"
        data-articleid="1286721"
        data-deleteurl="/delete-favorite"
        data-key="1286721"
        data-title="Amsterdam doet ertoe"
        data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
        id="favorite1286721"
        title="Bewaren"
      >
        <i
          className="icon-bookmark"
          title="Bewaren"
        />
        <i
          className="icon-bookmark1"
          title="Verwijder van bewaarlijst"
        />
        <span
          className="icon-label"
        >
          Bewaren
        </span>
      </span>
      <a
        className="comments-link"
        data-articleid="1286721"
        href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe#comments"
      >
        <i
          className="icon-comment-alt"
        />
        <span
          className="comments-count"
        >
          1
        </span>
        <span
          className="comments-suffix"
        >
          reactie
        </span>
      </a>
    </menu>
  </footer>
</div>
`;

exports[`OpeningTeaser renders multiple comments correctly 1`] = `
<div
  cardStyle="default"
  className="fd-card fd-opening-teaser default"
  id="1286721"
>
  <a
    href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
  >
    <figure
      className="fd-teaser-figure"
    >
      <picture>
        <source
          media="(max-width:640px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(max-width:860px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=723&h=406&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(min-width:861px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=719&h=404&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <img
          alt="Mario Draghi, president of the European Central Bank (ECB), departs a rates decision news conference in Frankfurt, Germany, on Thursday, Dec. 13, 2018. Draghi??said risks to the euro-area economy are worsening even as he called time on the??European Central Bank???s??flagship deflation-fighting tool. Photographer: Alex Kraus/Bloomberg"
          src="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
      </picture>
    </figure>
    <div
      className="meta"
    >
      <span
        className="prefix"
      >
        Buitenland
      </span>
      <time>
        34 min
      </time>
    </div>
    <div
      className="text-container"
    >
      <h1
        className="fd-typography opening-teaser-h"
        textStyle="opening-teaser-h"
      >
        Amsterdam doet ertoe
      </h1>
      <p
        className="intro"
      >
        Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.
      </p>
    </div>
  </a>
  <nav
    className="related"
  >
    <a
      className="related-link longread"
      href="https://dev.fd.nl/profiel/1235280/als-vrouw-moet-je-onafhankelijk-zijn"
    >
      <span
        className="related-title"
      >
        "Als vrouw moet je altijd onafhankelijk zijn"
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
    <a
      className="related-link"
      href="https://dev.fd.nl/economie-politiek/1235274/president-van-de-amerikaanse-fed-hamert-op-onafhankelijkheid"
    >
      <span
        className="related-title"
      >
        President van de Amerikaanse Fed hamert op onafhankelijkheid
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
  </nav>
  <footer
    className="fd-teaser-footer"
  >
    <menu>
      <span
        className="placeholder"
      />
      <span
        className="share-social"
      >
        <i
          className="icon-share-alt"
          data-id="1286721"
          data-section="Buitenland"
          data-title="Amsterdam doet ertoe"
          data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
          title="Delen"
        />
        <span
          className="icon-label"
        >
          Delen
        </span>
      </span>
      <span
        className="favorite"
        data-addurl="/add-favorite"
        data-articleid="1286721"
        data-deleteurl="/delete-favorite"
        data-key="1286721"
        data-title="Amsterdam doet ertoe"
        data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
        id="favorite1286721"
        title="Bewaren"
      >
        <i
          className="icon-bookmark"
          title="Bewaren"
        />
        <i
          className="icon-bookmark1"
          title="Verwijder van bewaarlijst"
        />
        <span
          className="icon-label"
        >
          Bewaren
        </span>
      </span>
      <a
        className="comments-link"
        data-articleid="1286721"
        href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe#comments"
      >
        <i
          className="icon-comment-alt"
        />
        <span
          className="comments-count"
        >
          99
        </span>
        <span
          className="comments-suffix"
        >
          reacties
        </span>
      </a>
    </menu>
  </footer>
</div>
`;

exports[`OpeningTeaser renders persoonlijk correctly 1`] = `
<div
  cardStyle="persoonlijk"
  className="fd-card fd-opening-teaser persoonlijk"
  id="1286721"
>
  <a
    href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
  >
    <figure
      className="fd-teaser-figure"
    >
      <picture>
        <source
          media="(max-width:640px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(max-width:860px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=723&h=406&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(min-width:861px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=719&h=404&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <img
          alt="Mario Draghi, president of the European Central Bank (ECB), departs a rates decision news conference in Frankfurt, Germany, on Thursday, Dec. 13, 2018. Draghi??said risks to the euro-area economy are worsening even as he called time on the??European Central Bank???s??flagship deflation-fighting tool. Photographer: Alex Kraus/Bloomberg"
          src="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
      </picture>
    </figure>
    <div
      className="meta"
    >
      <span
        className="prefix"
      >
        Buitenland
      </span>
      <time>
        34 min
      </time>
    </div>
    <div
      className="text-container"
    >
      <h1
        className="fd-typography opening-teaser-h"
        textStyle="opening-teaser-h"
      >
        Amsterdam doet ertoe
      </h1>
      <p
        className="intro"
      >
        Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.
      </p>
    </div>
  </a>
  <nav
    className="related"
  >
    <a
      className="related-link longread"
      href="https://dev.fd.nl/profiel/1235280/als-vrouw-moet-je-onafhankelijk-zijn"
    >
      <span
        className="related-title"
      >
        "Als vrouw moet je altijd onafhankelijk zijn"
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
    <a
      className="related-link"
      href="https://dev.fd.nl/economie-politiek/1235274/president-van-de-amerikaanse-fed-hamert-op-onafhankelijkheid"
    >
      <span
        className="related-title"
      >
        President van de Amerikaanse Fed hamert op onafhankelijkheid
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
  </nav>
  <footer
    className="fd-teaser-footer"
  >
    <menu>
      <span
        className="placeholder"
      />
      <span
        className="share-social"
      >
        <i
          className="icon-share-alt"
          data-id="1286721"
          data-section="Buitenland"
          data-title="Amsterdam doet ertoe"
          data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
          title="Delen"
        />
        <span
          className="icon-label"
        >
          Delen
        </span>
      </span>
      <span
        className="favorite"
        data-addurl="/add-favorite"
        data-articleid="1286721"
        data-deleteurl="/delete-favorite"
        data-key="1286721"
        data-title="Amsterdam doet ertoe"
        data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
        id="favorite1286721"
        title="Bewaren"
      >
        <i
          className="icon-bookmark"
          title="Bewaren"
        />
        <i
          className="icon-bookmark1"
          title="Verwijder van bewaarlijst"
        />
        <span
          className="icon-label"
        >
          Bewaren
        </span>
      </span>
      <a
        className="comments-link"
        data-articleid="1286721"
        href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe#comments"
      >
        <i
          className="icon-comment-alt"
        />
        <span
          className="comments-count"
        >
          1
        </span>
        <span
          className="comments-suffix"
        >
          reactie
        </span>
      </a>
    </menu>
  </footer>
</div>
`;

exports[`OpeningTeaser renders persoonlijk longread correctly 1`] = `
<div
  cardStyle="persoonlijk longread"
  className="fd-card fd-opening-teaser persoonlijk longread"
  id="1286721"
>
  <a
    href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
  >
    <figure
      className="fd-teaser-figure"
    >
      <picture>
        <source
          media="(max-width:640px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(max-width:860px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=723&h=406&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <source
          media="(min-width:861px)"
          srcSet="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=719&h=404&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
        <img
          alt="Mario Draghi, president of the European Central Bank (ECB), departs a rates decision news conference in Frankfurt, Germany, on Thursday, Dec. 13, 2018. Draghi??said risks to the euro-area economy are worsening even as he called time on the??European Central Bank???s??flagship deflation-fighting tool. Photographer: Alex Kraus/Bloomberg"
          src="https://images.fd.nl/-4mOGVA1ry0LHTzjvcxdyT-FH-E.jpg?fit=crop&crop=faces&auto=format&fm=jpg&q=45&w=599&h=337&rect=.0%2C.1519699812382739%2C.9999999999999999%2C.8442776735459663"
        />
      </picture>
    </figure>
    <div
      className="meta"
    >
      <span
        className="prefix"
      >
        Buitenland
      </span>
      <time>
        34 min
      </time>
    </div>
    <div
      className="text-container"
    >
      <h1
        className="fd-typography opening-teaser-h"
        textStyle="opening-teaser-h"
      >
        Amsterdam doet ertoe
      </h1>
      <p
        className="intro"
      >
        Een wethouder Verkeer en Vervoer, in Davos? Zeker! Wethouder Sharon Dijksma legt op het World Economic Forum uit hoe Amsterdam omgaat met duurzame mobiliteit.
      </p>
    </div>
  </a>
  <nav
    className="related"
  >
    <a
      className="related-link longread"
      href="https://dev.fd.nl/profiel/1235280/als-vrouw-moet-je-onafhankelijk-zijn"
    >
      <span
        className="related-title"
      >
        "Als vrouw moet je altijd onafhankelijk zijn"
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
    <a
      className="related-link"
      href="https://dev.fd.nl/economie-politiek/1235274/president-van-de-amerikaanse-fed-hamert-op-onafhankelijkheid"
    >
      <span
        className="related-title"
      >
        President van de Amerikaanse Fed hamert op onafhankelijkheid
      </span>
      <i
        className="icon-chevron-right"
      />
    </a>
  </nav>
  <footer
    className="fd-teaser-footer"
  >
    <menu>
      <span
        className="placeholder"
      />
      <span
        className="share-social"
      >
        <i
          className="icon-share-alt"
          data-id="1286721"
          data-section="Buitenland"
          data-title="Amsterdam doet ertoe"
          data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
          title="Delen"
        />
        <span
          className="icon-label"
        >
          Delen
        </span>
      </span>
      <span
        className="favorite"
        data-addurl="/add-favorite"
        data-articleid="1286721"
        data-deleteurl="/delete-favorite"
        data-key="1286721"
        data-title="Amsterdam doet ertoe"
        data-url="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe"
        id="favorite1286721"
        title="Bewaren"
      >
        <i
          className="icon-bookmark"
          title="Bewaren"
        />
        <i
          className="icon-bookmark1"
          title="Verwijder van bewaarlijst"
        />
        <span
          className="icon-label"
        >
          Bewaren
        </span>
      </span>
      <a
        className="comments-link"
        data-articleid="1286721"
        href="https://fd.nl/economie-politiek/1286721/amsterdam-doet-er-toe#comments"
      >
        <i
          className="icon-comment-alt"
        />
        <span
          className="comments-count"
        >
          1
        </span>
        <span
          className="comments-suffix"
        >
          reactie
        </span>
      </a>
    </menu>
  </footer>
</div>
`;
